### PR TITLE
Poc doctrine migrations

### PR DESCRIPTION
**Pull request by LukasReschke**
From https://github.com/owncloud/core/pull/14851 – needs extensive testing … 

Requires https://github.com/nextcloud/3rdparty/pull/24

### CODE REVIEW

File: core/Command/Db/Migrations/ExecuteCommand.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+use Symfony\Component\Console\Output\OutputInterface;
+
+class ExecuteCommand extends DBALExecuteCommand {
+
+	/** @var Connection */
+	private $ocConnection;
+
+	/**
+	 * @param \OCP\IConfig $config
+	 */
```
Comment (LukasReschke) [inline core/Command/Db/Migrations/ExecuteCommand.php]: PHPDoc should be adjusted.

File: core/Command/Db/Migrations/GenerateCommand.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+use Symfony\Component\Console\Output\OutputInterface;
+
+class GenerateCommand extends DBALGenerateCommand {
+
+	/** @var Connection */
+	private $ocConnection;
+
+	/**
+	 * @param \OCP\IConfig $config
+	 */
```
Comment (LukasReschke) [inline core/Command/Db/Migrations/GenerateCommand.php]: PHPDoc should be adjusted

File: core/Command/Db/Migrations/MigrateCommand.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+use Symfony\Component\Console\Output\OutputInterface;
+
+class MigrateCommand extends DBALMigrateCommand {
+
+	/** @var Connection */
+	private $ocConnection;
+
+	/**
+	 * @param \OCP\IConfig $config
+	 */
```
Comment (LukasReschke) [inline core/Command/Db/Migrations/MigrateCommand.php]: PHPDoc should be adjusted

File: core/Command/Db/Migrations/StatusCommand.php [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+use Symfony\Component\Console\Output\OutputInterface;
+
+class StatusCommand extends DBALStatusCommand {
+
+	/** @var Connection */
+	private $ocConnection;
+
+	/**
+	 * @param \OCP\IConfig $config
+	 */
```
Comment (LukasReschke) [inline core/Command/Db/Migrations/StatusCommand.php]: PHPDoc should be adjusted

File: lib/private/DB/MDB2SchemaReader.php [hunk 5]
```diff
@@ -80,8 +80,10 @@ public function __construct(IConfig $config, AbstractPlatform $platform) {
 	 * @return \Doctrine\DBAL\Schema\Schema
 	 * @throws \DomainException
 	 */
-	public function loadSchemaFromFile($file) {
-		$schema = new \Doctrine\DBAL\Schema\Schema();
+	public function loadSchemaFromFile($file, $schema = null) {
```
Comment (LukasReschke) [inline lib/private/DB/MDB2SchemaReader.php]: $schema is not in PHPDoc

File: lib/private/Setup/AbstractDatabase.php [hunk 6]
```diff
@@ -26,8 +26,12 @@
  */
 namespace OC\Setup;
 
+<<<<<<< HEAD
 use OC\AllConfig;
 use OC\DB\ConnectionFactory;
+=======
```
Comment (LukasReschke) [inline lib/private/Setup/AbstractDatabase.php]: Merge fuckup :)

File: core/Command/Db/Migrations/ExecuteCommand.php [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+	public function __construct(IConfig $config, Connection $connection) {
+		$this->config = $config;
```
Comment (LukasReschke) [inline core/Command/Db/Migrations/ExecuteCommand.php]: Unused

**Comment by LukasReschke** [comment]
Also there is https://github.com/owncloud/activity/pull/546 which uses this approach.

FYI @nickvergessen

**Comment by MorrisJobke** [comment]
> From owncloud/core#14851 – needs extensive testing …

Yep :) I would say we should test this with a first real world example for those migrations and get this in as fast as possible. Then do all our migrations with this tool.

**Review by MorrisJobke — COMMENTED** [review]
Looks good so far - 3rdparty needs to be fixed

**Comment by LukasReschke** [comment]
Closing this as upstream went with another approach and we should reconsider on how we want to proceed here. We can reopen at any time anyhow…

**Comment by MorrisJobke** [comment]
> Closing this as upstream went with another approach and we should reconsider on how we want to proceed here. We can reopen at any time anyhow…

I would like to go the doctrine migrations approach.

**Pull request closed without merging**